### PR TITLE
Bid strategy depends on medium

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: app/Http/Controllers/Manager/BidStrategyController.php [hunk 1]
```diff
@@ -243,58 +208,52 @@ private static function processTargetingOptionValues(
...
 
-    public function putBidStrategy(BidStrategyRequest $request): JsonResponse
+    private static function prepareOptionForInputType(array $bidStrategyDetails, string $parentKey): array
+    {
+        $optionValues = [
+            '' => 'MISSING',
+            '*' => 'DEFAULT',
+        ];
+        foreach ($bidStrategyDetails as $key => $value) {
+            $id_parts = explode(':', $key);
```
Comment (m-pilarczyk) [inline app/Http/Controllers/Manager/BidStrategyController.php]:
```suggestion
            $idParts = explode(':', $key);
```

File: app/Http/Controllers/Manager/BidStrategyController.php [hunk 2]
```diff
@@ -243,58 +208,52 @@ private static function processTargetingOptionValues(
...
+            $id = array_pop($id_parts);
```
Comment (m-pilarczyk) [inline app/Http/Controllers/Manager/BidStrategyController.php]:
```suggestion
            $id = array_pop($idParts);
```

File: app/Http/Controllers/Manager/BidStrategyController.php [hunk 3]
```diff
@@ -243,58 +208,52 @@ private static function processTargetingOptionValues(
...
+            $prefix = implode(':', $id_parts);
```
Comment (m-pilarczyk) [inline app/Http/Controllers/Manager/BidStrategyController.php]:
```suggestion
            $prefix = implode(':', $idParts);
```

**Review by m-pilarczyk — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Manager/BidStrategyController.php (hunk 2 above)
Comment (PawelPodkalicki) [reply app/Http/Controllers/Manager/BidStrategyController.php]: Variable renamed

File: app/Http/Controllers/Manager/BidStrategyController.php (hunk 3 above)
Comment (PawelPodkalicki) [reply app/Http/Controllers/Manager/BidStrategyController.php]: Variable renamed

File: app/Http/Controllers/Manager/BidStrategyController.php (hunk 1 above)
Comment (PawelPodkalicki) [reply app/Http/Controllers/Manager/BidStrategyController.php]: Variable renamed

**Review by m-pilarczyk — APPROVED** [review]

**Pull request merged**